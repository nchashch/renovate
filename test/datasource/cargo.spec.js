const got = require('got');
const { getPkgReleases } = require('../../lib/datasource/cargo');
const fs = require('fs');

let res1 = fs.readFileSync('test/_fixtures/cargo/libc.json', 'utf8');
res1 = JSON.parse(res1);

jest.mock('got');

describe('datasource/cargo', () => {
  describe('getPkgReleases', () => {
    it('returns null for empty result', async () => {
      got.mockReturnValueOnce(null);
      expect(
        await getPkgReleases('non_existent_crate')
      ).toBeNull();
    });
    it('returns null for 404', async () => {
      got.mockImplementationOnce(() =>
                                 Promise.reject({
                                   statusCode: 404,
                                 })
                                );
      expect(
        await getPkgReleases('some_crate')
      ).toBeNull();
    });
    it('returns null for unknown error', async () => {
      got.mockImplementationOnce(() => {
        throw new Error();
      });
      expect(
        await getPkgReleases('some_crate')
      ).toBeNull();
    });
    it('processes real data', async () => {
      got.mockReturnValueOnce({
        body: res1,
      });
      const res = await getPkgReleases({fullname: 'libc'});
      expect(res).toMatchSnapshot();
      expect(res).not.toBeNull();
      expect(res).toBeDefined();
    });
    it('skips wrong package', async () => {
      got.mockReturnValueOnce({
        body: res1,
      });
      const res = await getPkgReleases('wrong_crate');
      expect(res).toBeNull();
    });
  });
});
