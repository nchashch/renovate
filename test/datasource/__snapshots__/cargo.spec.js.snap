// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datasource/cargo getPkgReleases processes real data 1`] = `
Object {
  "releases": Array [
    "0.2.44",
  ],
  "repository": "https://github.com/rust-lang/libc",
}
`;

exports[`datasource/cargo getPkgReleases processes real data 2`] = `
Object {
  "homepage": "https://www.amethyst.rs/",
  "releases": Array [
    "0.9.0",
  ],
  "repository": "https://github.com/amethyst/amethyst",
}
`;
